lib/opam_solve.ml

```diff
@@ -175,6 +178,22 @@ module Opam_monorepo_context (Base_context : BASE_CONTEXT) :
     in
     regular_versions @ avoid_versions
 
+  let prefer_version ~version versions =
+    let rec aux acc l =
```

Comment:
`aux` is a rather nondescript name and given the function is somewhat dense (it is not a loop) I would try to give it a better name. Also `l` could be replaced by `function`.

Reply:
I personnaly quite dislike `function`, I prefer to declare all of my arguments so if you don't mind I'll stick to this.

The `aux` function here is just a recursive version of the main function with an extra `accumulator` arg. I find it to be a quite common pattern but I can rename it `recursive_prefer_version` if you'd like.

Reply:
It is more like `move_version_to_front`, because that describes what it is doing.

It is fine to dislike `function`, I can see the point because sometimes naming the arguments really helps, but in such case it would be good if the argument was called something else than `l`.